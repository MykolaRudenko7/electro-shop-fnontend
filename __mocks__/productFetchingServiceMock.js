export const productFetchingServiceMock = {
  productsSuccessResponse: {
    data: [
      {
        inStock: true,
        slug: 'mi-lite-15',
        url: 'https://res.cloudinary.com/dqdonqby4/image/upload/v1612476949/new-products/1_hw0qj0.jpg',
        name: 'Laptop Mi Notebook Lite 15 I5/8/512/MX110/W (JYU4139CN)',
        reviewsCount: 10,
        previousPrice: 29999,
        rating: 4.3,
        price: 21499,
      },
    ],
  },

  laptopsErrorResponse: {
    status: 400,
    statusText: 'Bad Request',
    data: { message: 'Laptops not found' },
  },

  newProductsErrorResponse: {
    status: 400,
    statusText: 'Bad Request',
    data: { message: 'New Products not found' },
  },
}
