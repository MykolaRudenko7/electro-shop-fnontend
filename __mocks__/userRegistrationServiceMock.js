const {
  TEST_EMAIL,
  TEST_PASSWORD,
  TEST_SIGN_UP_NAME,
  TEST_SIGN_UP_MOBILE_NUMBER,
  TEST_REFRESH_TOKEN,
  TEST_ACCESS_TOKEN,
  TEST_HASH_PASSWORD,
  TEST_USER_ID,
  TEST_ACTIVATION_LINK,
} = process.env

export const userRegistrationServiceMock = {
  userLoginResponse: {
    data: [
      {
        accessToken: TEST_ACCESS_TOKEN,
        message: 'Log in successfully',
        refreshToken: TEST_REFRESH_TOKEN,
        success: true,
        user: {
          activationlink: TEST_ACTIVATION_LINK,
          email: TEST_EMAIL,
          isActivated: false,
          lockUntil: null,
          loginAttempts: 0,
          mobileNumber: TEST_SIGN_UP_MOBILE_NUMBER,
          name: TEST_SIGN_UP_NAME,
          password: TEST_HASH_PASSWORD,
          __v: 0,
          _id: TEST_USER_ID,
        },
      },
    ],
    status: 200,
  },
  signUpUserRequest: {
    name_newUser: TEST_SIGN_UP_NAME,
    email_newUser: TEST_PASSWORD,
    password_newUser: TEST_PASSWORD,
    confirmPassword_newUser: TEST_PASSWORD,
    mobileNumber_newUser: TEST_SIGN_UP_MOBILE_NUMBER,
  },
  badSignUpUserRequest: {
    name_newUser: TEST_SIGN_UP_NAME,
    email_newUser: 'wrongEmail@gmail.com',
    password_newUser: 'wrongPassword',
    confirmPassword_newUser: 'wrongPassword',
    mobileNumber_newUser: '+wrongNumber',
  },
  signInUserRequest: {
    email_signIn: TEST_EMAIL,
    password_signIn: TEST_PASSWORD,
  },
  badSignInUserRequest: {
    email_signIn: TEST_EMAIL,
    password_signIn: 'wrongPassword',
  },
  logOutUserResponse: { success: true, message: 'Log out successfully' },

  errorSignInResponse: { status: 400, message: 'User with email does not exist' },
  errorSignUpResponse: { status: 400, message: 'User with this email or phone already exists' },
  errorLogOutResponse: { status: 400, message: 'Network Error' },
  unauthorizedErrorResponse: { status: 401, message: 'User is not authorized' },
}
