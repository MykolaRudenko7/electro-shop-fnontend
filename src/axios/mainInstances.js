import Axios from 'axios'
import { setupCache } from 'axios-cache-interceptor'

const { API_BASE_URL } = process.env

const mainApiClient = Axios.create({
  baseURL: API_BASE_URL,
})

export const axiosInstanceCache = setupCache(mainApiClient)
