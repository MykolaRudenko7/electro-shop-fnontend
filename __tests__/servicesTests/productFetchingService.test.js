import axios from 'axios'
import ProductsService from 'src/services/productFetchingService'
import { productFetchingServiceMock } from '__mocks__/productFetchingServiceMock'

const { productsSuccessResponse, laptopsErrorResponse, newProductsErrorResponse } =
  productFetchingServiceMock

jest.mock('axios', () => ({
  get: jest.fn(),
  create: jest.fn(() => axios),
}))

describe('ProductFetchingService API requests', () => {
  it('Should return laptops data on successful request', async () => {
    axios.get = jest.fn().mockResolvedValue(productsSuccessResponse)

    const laptops = await ProductsService.fetchLaptops()

    expect(axios.get).toHaveBeenCalledTimes(1)
    expect(laptops).toEqual(productsSuccessResponse.data)
    expect(laptops).toMatchSnapshot()
  })

  it('Should return an object with error status and message if laptops request is incorrect', async () => {
    axios.get.mockImplementationOnce(() => Promise.reject(laptopsErrorResponse))

    await expect(ProductsService.fetchLaptops()).rejects.toMatchObject(laptopsErrorResponse)
  })

  it('Should return new products data on successful request', async () => {
    axios.get = jest.fn().mockResolvedValue(productsSuccessResponse)

    const newProducts = await ProductsService.fetchNewProducts()

    expect(axios.get).toHaveBeenCalledTimes(1)
    expect(newProducts).toEqual(productsSuccessResponse.data)
    expect(newProducts).toMatchSnapshot()
  })

  it('Should return an object with error status and message if new products request is incorrect', async () => {
    axios.get.mockImplementationOnce(() => Promise.reject(newProductsErrorResponse))

    await expect(ProductsService.fetchLaptops()).rejects.toMatchObject(newProductsErrorResponse)
  })
})
