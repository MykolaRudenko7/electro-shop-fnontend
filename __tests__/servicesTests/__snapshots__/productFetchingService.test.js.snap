// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductFetchingService API requests Should fetch laptops successfully 1`] = `
[
  {
    "inStock": true,
    "name": "Laptop Mi Notebook Lite 15 I5/8/512/MX110/W (JYU4139CN)",
    "previousPrice": 29999,
    "price": 21499,
    "rating": 4.3,
    "reviewsCount": 10,
    "slug": "mi-lite-15",
    "url": "https://res.cloudinary.com/dqdonqby4/image/upload/v1612476949/new-products/1_hw0qj0.jpg",
  },
]
`;

exports[`ProductFetchingService API requests Should fetch newProducts successfully 1`] = `
[
  {
    "inStock": true,
    "name": "Laptop Mi Notebook Lite 15 I5/8/512/MX110/W (JYU4139CN)",
    "previousPrice": 29999,
    "price": 21499,
    "rating": 4.3,
    "reviewsCount": 10,
    "slug": "mi-lite-15",
    "url": "https://res.cloudinary.com/dqdonqby4/image/upload/v1612476949/new-products/1_hw0qj0.jpg",
  },
]
`;

exports[`ProductFetchingService API requests Should return laptops data on successful request 1`] = `
[
  {
    "inStock": true,
    "name": "Laptop Mi Notebook Lite 15 I5/8/512/MX110/W (JYU4139CN)",
    "previousPrice": 29999,
    "price": 21499,
    "rating": 4.3,
    "reviewsCount": 10,
    "slug": "mi-lite-15",
    "url": "https://res.cloudinary.com/dqdonqby4/image/upload/v1612476949/new-products/1_hw0qj0.jpg",
  },
]
`;

exports[`ProductFetchingService API requests Should return new products data on successful request 1`] = `
[
  {
    "inStock": true,
    "name": "Laptop Mi Notebook Lite 15 I5/8/512/MX110/W (JYU4139CN)",
    "previousPrice": 29999,
    "price": 21499,
    "rating": 4.3,
    "reviewsCount": 10,
    "slug": "mi-lite-15",
    "url": "https://res.cloudinary.com/dqdonqby4/image/upload/v1612476949/new-products/1_hw0qj0.jpg",
  },
]
`;
